src/features/klant/contact/service.ts

```diff
@@ -0,0 +1,127 @@
+import {
+  ServiceResult,
+  fetchLoggedIn,
+  parsePagination,
+  throwIfNotOk,
+  parseJson,
+} from "@/services";
+import { mutate } from "swrv";
+import type { Ref } from "vue";
+import { KlantType, type Klant } from "../types";
+
+const klantenBaseUrl = "/api/klanten/api/v1/klanten";
+
+type KlantSearchParameters = {
+  email?: string;
+  phone?: string;
+};
+
+const klantRootUrl = new URL(document.location.href);
+klantRootUrl.pathname = klantenBaseUrl;
+
+function getKlantSearchUrl({ email = "", phone = "" }: KlantSearchParameters) {
+  if (!email && !phone) return "";
+
+  const url = new URL(klantRootUrl);
+
+  url.searchParams.set("subjectType", KlantType.Persoon);
+
+  if (email) {
+    url.searchParams.set("emailadres", email);
+  }
+
+  if (phone) {
+    url.searchParams.set("telefoonnummer", phone);
+  }
+
+  return url.toString();
+}
+
+function mapKlant(obj: any): Klant {
```

Comment:
is het verstandig hier zo'n algemene mapping met een brede Klant interface te gebruiken? wellicht een specifieker maken (een Contact interface of type, mapContact, fetchContactById, ..)

Reply:
Dit is een goed aandachtspunt, dat we denk ik makkelijker kunnen tackelen als we de rest van de klantbeeldwijzigingen ook doorgevoerd hebben. Er moet mijn inziens namelijk nog wel iets van een gemene deler zijn tussen klanten, met name voor de contactmomentstore en het afhandelingsscherm. Ik zet er een comment bij.